oio/xcute/common/backend.py

```diff
@@ -506,50 +494,41 @@ def create(self, job_type, job_config):
         job_id = datetime.utcnow().strftime('%Y%m%d%H%M%S%f') \
             + '-%011x' % random.randrange(16**11)
 
+        _job_config = dict()
+        for key, value in job_config.items():
+            _job_config['config.' + key] = value
+        _job_config['config.params'] = json.dumps(
+            _job_config['config.params'])
+
         self.script_create(
             keys=[self._get_timestamp(), job_id, job_type],
-            args=self._dict_to_lua_array(
-                self._marshal_job_config(job_config)),
+            args=self._dict_to_lua_array(_job_config),
             client=self.conn)
         return job_id
 
-    @handle_redis_exceptions
-    def start_job(self, job_id, job_info):
-        pipeline = self.conn.pipeline()
-
-        self._update_job_info(job_id, job_info, client=pipeline)
-
-        pipeline.execute()
-
     def list_orchestrator_jobs(self, orchestrator_id):
         orchestrator_jobs_key = self.key_orchestrator_jobs % orchestrator_id
         job_ids = self.conn.smembers(orchestrator_jobs_key)
 
         pipeline = self.conn.pipeline()
         for job_id in job_ids:
-            self._get_job_conf(job_id, client=pipeline)
             self._get_job_info(job_id, client=pipeline)
-        job_config_infos = pipeline.execute()
+        job_infos = pipeline.execute()
 
-        return (
-            (job_id, self._unmarshal_job_config(job_config),
-             self._unmarshal_job_info(job_info))
-            for job_config, job_info in zip(*([iter(job_config_infos)] * 2))
-        )
+        return (self._unmarshal_job_info(job_info)
+                for job_info in job_infos)
```

Comment:
If we don't shun `map`, `return map(self._unmarshal_job_info, job_infos)` :smiley:

Reply:
Well seen.

Reply:
It was missing a check if the job was deleted between two. I can not use `map` anymore ...